src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer_ReleaseTracking.cs

```diff
@@ -0,0 +1,900 @@
+                isEnabledByDefault != null && isEnabledByDefault != releaseTrackingLine.EnabledByDefault ||
+                defaultSeverity != null && defaultSeverity != releaseTrackingLine.DefaultSeverity)
+            {
+                string propertyName;
+                (string category, bool? isEnabledByDefault, DiagnosticSeverity? defaultSeverity)? oldRule = null;
+                if (!releaseTrackingLine.IsShipped)
+                {
+                    // For existing entry in unshipped file, we can just update.
+                    propertyName = EntryToUpdatePropertyName;
+                    if (releaseTrackingLine is ChangedRuleReleaseTrackingLine changedLine)
+                    {
+                        oldRule = (changedLine.OldCategory, changedLine.OldEnabledByDefault, changedLine.OldDefaultSeverity);
+                    }
+                }
+                else
+                {
+                    // Need to add a new changed rule entry in unshipped file.
+                    propertyName = EntryToAddPropertyName;
+                    oldRule = (releaseTrackingLine.Category, releaseTrackingLine.EnabledByDefault, releaseTrackingLine.DefaultSeverity);
+                }
+
+                var newEntry = GetEntry(ruleId, category, helpLink, isEnabledByDefault, defaultSeverity, oldRule);
+                var properties = ImmutableDictionary<string, string?>.Empty.Add(propertyName, newEntry);
+                var diagnostic = ruleIdArgument.CreateDiagnostic(UpdateDiagnosticIdInAnalyzerReleaseRule, properties, ruleId);
+                addDiagnostic(diagnostic);
+                return;
+            }
+        }
+
+        private static string GetEntry(
+            string ruleId,
+            string? category,
+            string? helpLink,
+            bool? isEnabledByDefault,
+            DiagnosticSeverity? defaultSeverity,
+            (string category, bool? isEnabledByDefault, DiagnosticSeverity? defaultSeverity)? oldRule = null)
+        {
+            // Rule ID | Category | Severity | HelpLink (optional)
+            //      OR
+            // Rule ID | New Category | New Severity | Old Category | Old Severity | HelpLink (optional)
+            var entry = $"{ruleId} | {GetCategoryText(category)} | {GetSeverityText(isEnabledByDefault, defaultSeverity)} |";
+
+            if (oldRule.HasValue)
+            {
+                entry += $" {GetCategoryText(oldRule.Value.category)} | {GetSeverityText(oldRule.Value.isEnabledByDefault, oldRule.Value.defaultSeverity)} |";
+            }
+
+            helpLink ??= TryGetHelpLinkForCARule(ruleId);
+            if (!string.IsNullOrEmpty(helpLink))
+            {
+                entry += $" [Documentation]({helpLink})";
+            }
+
+            return entry;
+
+            static string GetCategoryText(string? category)
+                => category ?? UndetectedText;
+
```

Comment:
Totally an edge case, but are there any restrictions on category names not containing new lines and pipes?

Reply:
Hmm, I haven’t seen multiline categories in any package. We should consider those as unsupported for all practical purposes.